config/config-api/src/com/thoughtworks/go/config/EnvironmentVariableConfig.java

```diff
@@ -339,4 +348,18 @@ public void setEntityId(Long entityId) {
     public void setEntityType(String entityType) {
         this.entityType = entityType;
     }
+
+    @Override
+    public ConfigOrigin getOrigin() {
+        return origin;
+    }
+
+    public boolean isRemote()
+    {
+        return origin != null && !origin.isLocal();
```

Comment:
So you mean we could have something like
```
private ConfigOrigin origin = new NullOrigin();
```
in all classes that have `origin` field?
I think it might not work in all cases, e.g. magical loader instantiates elements by reflection, it would be null then. But we can try. Do you want a commit added in this PR or should we refactor with your changes too?